web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -258,29 +259,34 @@ public void contextInitialized(ServletContextEvent event) {
 			log.error(MarkerFactory.getMarker("FATAL"), "Failed to obtain JDBC connection", e);
 		}
 	}
-	
-	private void loadCsrfGuardProperties(ServletContext servletContext) throws FileNotFoundException, IOException {	
-		File file = new File(OpenmrsUtil.getApplicationDataDirectory(), "csrfguard.properties");
-		InputStream inputStream = null;
+
+	private void loadCsrfGuardProperties(ServletContext servletContext) throws IOException {
```

Comment:
Note that `loadCsrfGuardProperties()` is currently called _before_ the runtime properties are setup, which means `OpenmrsUtil.getApplicationDirectory()` (inter alia) can return the wrong folder. Ideally, this call should be moved to after the runtime properties are loaded.